pkg/kn/commands/service_list.go

```diff
@@ -44,22 +45,42 @@ func NewServiceListCommand(p *KnParams) *cobra.Command {
 				return err
 			}
 
-			printer, err := serviceListPrintFlags.ToPrinter()
-			if err != nil {
-				return err
-			}
 			service.GetObjectKind().SetGroupVersionKind(schema.GroupVersionKind{
 				Group:   "knative.dev",
 				Version: "v1alpha1",
 				Kind:    "Service"})
-			err = printer.PrintObj(service, cmd.OutOrStdout())
-			if err != nil {
+
+			printer := printers.GetNewTabWriter(cmd.OutOrStdout())
+
+			if err := printServiceList(printer, *service); err != nil {
 				return err
 			}
+			printer.Flush()
+
 			return nil
 		},
 	}
 	AddNamespaceFlags(serviceListCommand.Flags(), true)
-	serviceListPrintFlags.AddFlags(serviceListCommand)
 	return serviceListCommand
 }
+
+func printServiceList(printer *tabwriter.Writer, services servingv1alpha1.ServiceList) error {
+	// case where no services are present
+	if len(services.Items) < 1 {
+		fmt.Fprintln(printer, "No resources found.")
+		return nil
+	}
+	columnNames := []string{"NAME", "DOMAIN", "LATESTCREATED", "LATESTREADY", "AGE"}
+	if _, err := fmt.Fprintf(printer, "%s\n", strings.Join(columnNames, "\t")); err != nil {
+		return err
+	}
+	for _, ksvc := range services.Items {
+		_, err := fmt.Fprintf(printer, "%s\n", strings.Join([]string{ksvc.Name, ksvc.Status.Domain,
+			ksvc.Status.LatestCreatedRevisionName, ksvc.Status.LatestReadyRevisionName,
+			printers.CalculateAge(ksvc.CreationTimestamp.Time)}, "\t"))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
You shouldn't forget to `printer.Flush()` to get the output in any case at that point.

Reply:
Right, `printer.Flush()` is invoked at https://github.com/knative/client/pull/56/files#diff-633efd289eb54521b1cdd9b31c7025c9R58 in calling function before final return.

Reply:
now using `defer`